### PR TITLE
4714 edit vax UI

### LINKED ISSUE #4714: Edit vaccinations
Implement edit of vaccinations

If batch or item changes, we will need to create a Inventory Addition to reverse the Prescription, and create a new Prescription.

If change from given to not given, need to create the Inventory Addition to reverse the Prescription.

If change from not given to given, create the prescription like in create vaccination

### PR DESCRIPTION
**Pull request by lache-melvin**
Fixes #4714

~~Uses both #4809 and #4802 for a base - will rebase once one of those is merged! Maybe wait for review until then? 🙏~~ Ready now :)

# 👩🏻‍💻 What does this PR do?

Wires up update of vaccinations from the vaccination modal.

Created/reversed prescriptions in the ledger:
Screenshot 2024-09-18 at 1 20 37 PM

## 💌 Any notes for the reviewer?

# 🧪 Testing

- [ ] From vaccine card, create a vaccination
- [ ] Click that line again, and edit some of the details
- [ ] Saving updates the vaccination
- [ ] Change from not given to given - confirm prescription is created
- [ ] Change from given to not given - confirm reverse inventory adjustment is created

# 📃 Documentation

- [x] **Part of an epic**: documentation will be completed for the feature as a whole
- [ ] **No documentation required**: no user facing changes or a bug fix which isn't a change in behaviour
- [ ] **These areas should be updated or checked**: 
  1.
  2.

### CODE REVIEW

**Comment by CarlosNZ** [comment]
> Uses both #4809 and #4802 for a base - will rebase once one of those is merged! Maybe wait for review until then?

Ah yes, I'm seeing my changes from #4802 in here too. Will wait till that is merged and we re-merge from develop to look at this properly.

**Comment by lache-melvin** [comment]
This should be ready for review now @CarlosNZ! 🙏

**Review by CarlosNZ — APPROVED** [review]
LGTM, Thanks :)

File: client/packages/system/src/Vaccination/api/useVaccination.ts [hunk 1]
```diff
@@ -159,6 +162,52 @@ const useInsert = ({
...
+      throw new Error(t('error.failed-to-save-vaccination'));
+    }
+
+    const apiResult = await api.updateVaccination({
+      storeId,
+      input: {
+        id: vaccinationId,
+        given: input.given ?? false,
+        vaccinationDate: Formatter.naiveDate(input.date ?? new Date()),
+        clinicianId: setNullableInput('id', input.clinician),
```
[the remark this answers is not in the document]
Comment (lache-melvin) [reply client/packages/system/src/Vaccination/api/useVaccination.ts:187]: Ah yeah, we use that in a few places where we want to distinguish between an `null` input meaning "set the value to null" and "don't update the value", but yeah a comment in there would help!

[the remark this answers is not in the document]
Comment (CarlosNZ) [reply client/packages/system/src/Vaccination/api/useVaccination.ts:187]: Ah yes of course-- that sounds familiar... 👀

Screenshot 2024-09-20 at 4 23 25 PM

**Pull request merged**